public/app/core/components/QueryOperationRow/QueryOperationRowHeader.tsx

```diff
@@ -33,10 +33,12 @@ export const QueryOperationRowHeader: React.FC<QueryOperationRowHeaderProps> = (
   return (
     <div className={styles.header}>
       <div className={styles.column}>
-        <Icon
+        <IconButton
           name={isContentVisible ? 'angle-down' : 'angle-right'}
+          title="Open and close query row"
```

Comment:
we should probably use `aria-label` instead of `title`.

also `Open and close query row` doesn't really describe the button, as clicking on it doesn't "open and close" the query row, but rather toggle its collapsed state.

FYI there are some more considerations for expandable content to take into account to make this really accessible (ie. https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-expanded) but maybe we should tackle it in a separate issue.

Reply:
As title and aria-label are not redundant I added the aria-label. Additionally I changed the text and checked it by using VoiceOver.

About the second part: I agree that this could be an option and before we decide to implement the aria-expanded attribute we should discuss a couple of things. What's the position of a11y-wg on that topic? Are there actually people using Grafana with screen readers? 
And if we implement it here we should consider implementing it in other places, e.g. query history.
I can open a draft issue for it and add the questions in the description. So we we can talk about it later.

Reply:
maybe my previous comment wasn't clear. this control doesn't toggle "collapse AND expand" property of the row, it toggle only one thing: it's "collapsed" state.

about the second part, that's the whole point of labelling elements, regardless of having screen reader users or not grafana should be accessible by everyone.

the original issue is about keyboard navigation as this control wasn't usable via keyboard, adding an aria label or a title just for the sake of it or implementing an half baked solution is not solving any accessibility issue.
if we want to improve accessibility for screen reader we should properly use the correct roles for this and related elements. ie https://www.w3.org/WAI/ARIA/apg/example-index/accordion/accordion.

in my opinion we should either fix it completely or not. i'd suggest to create a separate issue and describe what should be the a11y success criteria for a that button that controls a collapsable section and remove any code non relevant to the original issue from this PR.

Reply:
> maybe my previous comment wasn't clear. this control doesn't toggle "collapse AND expand" property of the row, it toggle only one thing: it's "collapsed" state.

I get what you are saying and I consciously decided which text I want to add for the title and aria-label. Both of these texts are meant to help our users (not necessarily developers) who are not interested in any state we handle in the background but need to know what the button does they are about to click.

> about the second part, that's the whole point of labelling elements, regardless of having screen reader users or not grafana should be accessible by everyone.

I totally agree and I am not questioning our efforts to make Grafana accessible at all.

> the original issue is about keyboard navigation as this control wasn't usable via keyboard, adding an aria label or a title just for the sake of it or implementing an half baked solution is not solving any accessibility issue.

Adding a title and an aria-label is not half baked considering that the button-like element I replaced was not implemented thoroughly. It should have had both from the very beginning. So leaving them away and adding them later seems to me like only doing half the job of implementing this button properly and I needed to implement the button to fix the issue with the keyboard a11y. Can you imagine a visually impaired person getting to that button and it doesn't even tell them what it does? This would be even more frustrating than not knowing whether the section below is expanded or not.

BTW I already opened an issue to take care of the collapsable section below the button.